### PR TITLE
Fix solid recursion bug

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Passes a `renderId` into solid so that we match up the render to hydration, in case there is multiple hydrations occuring on the page.
- Closes https://github.com/withastro/astro/issues/3947

## Testing

- e2e test added

## Docs

N/A, bug fix

### CODE REVIEW

File: packages/integrations/solid/src/client.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-			window._$HY = { events: [], completed: new WeakSet(), r: {} };
+		if (!(window as any)._$HY) {
+			(window as any)._$HY = { events: [], completed: new WeakSet(), r: {} };
 		}
 		if (!element.hasAttribute('ssr')) return;
 
 		const fn = client === 'only' ? render : hydrate;
 
-		let _slots = {};
+		console.log("I AM HERE");
```
Comment (natemoo-re) [inline packages/integrations/solid/src/client.ts:14]: Missed a `console.log`

File: packages/integrations/solid/src/client.ts [hunk 2]
```diff
@@ -28,6 +30,10 @@ export default (element) =>
 		}
 
 		const { default: children, ...slots } = _slots;
+		const renderId = element.dataset.solidRenderId;
+		console.log("SOLID RENDER ID", renderId);
+
+		document.documentElement.setAttribute('some-test', renderId + '-test')
```
Comment (natemoo-re) [inline packages/integrations/solid/src/client.ts:36]: And some debug code

**Review by natemoo-re — COMMENTED** [review]
Thanks for the fix!

File: packages/integrations/solid/src/client.ts (hunk 1 above)
Comment (matthewp) [reply packages/integrations/solid/src/client.ts:14]: hah!

File: packages/integrations/solid/src/client.ts (hunk 2 above)
Comment (matthewp) [reply packages/integrations/solid/src/client.ts:36]: oh wow, this is embarrasing.

**Review by natemoo-re — APPROVED** [review]
LGTM! I knew having renderers return an object was going to pay off one day 😅

**Pull request merged**